src/main/java/com/bytebreak/animagic/texture/AnimagicTextureRegion.java

```diff
@@ -0,0 +1,78 @@
+package com.bytebreak.animagic.texture;
+
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g2d.TextureRegion;
+import com.badlogic.gdx.math.Vector2;
+
+public class AnimagicTextureRegion extends TextureRegion {
+    private Texture normals;
+    private float offsetX = 0;
+    private float offsetY = 0;
```

Comment:
I was thinking we may want to abstract this one layer out and have an `AnimationData` class, potentially. It would make serialization of it even more trivial, and would allow for us to expand on it without breaking anything.